src/main.c

```diff
@@ -880,10 +880,42 @@ ubus_poe_reload_cb(struct ubus_context *ctx, struct ubus_object *obj,
 	return UBUS_STATUS_OK;
 }
 
+static const struct blobmsg_policy ubus_poe_port_policy[] = {
+	{ "port", BLOBMSG_TYPE_STRING },
+	{ "enable", BLOBMSG_TYPE_BOOL },
+};
+
+static int
+ubus_poe_port_cb(struct ubus_context *ctx, struct ubus_object *obj,
+		 struct ubus_request_data *req, const char *method,
+		 struct blob_attr *msg)
+{
+	struct blob_attr *tb[ARRAY_SIZE(ubus_poe_port_policy)];
+	char *port;
+	size_t i;
+
+	blobmsg_parse(ubus_poe_port_policy,
+		      ARRAY_SIZE(ubus_poe_port_policy),
+		      tb, blob_data(msg), blob_len(msg));
+	if (!tb[0] || !tb[1])
+		return UBUS_STATUS_INVALID_ARGUMENT;
+
+	port = blobmsg_get_string(tb[0]);
+	for (i = 0; i < config.port_count; i++) {
+		if (!config.ports[i].valid || strncmp(port, config.ports[i].name, sizeof(config.ports[i].name)))
+			continue;
+
+		poe_cmd_port_enable(i, !!blobmsg_get_bool(tb[1]));
+		return UBUS_STATUS_OK;
+	}
+	return UBUS_STATUS_NOT_FOUND;
```

Comment:
Does it matter?  The code becomes more complicated with this is reversed, since we will have to finish the loop and then test again whether it succeeded or not.  We can't decide that it failed inside the loop.

Not extremely important to me, but I just like making code as simple as possible.

Reply:
It's not critical. Either form will work for now. This is up to you.

> We can't decide that it failed inside the loop.

But of course!
```
	for (i = 0; i < config.port_count; i++) {
		port = &config.ports[i];
		if (port->enable && strcmp(port_name, port->name)
			break;

		port = NULL;
	}
	
	if (!port)
		return UBUS_STATUS_NOT_FOUND;
```